documentation/API_documentation/QoD_Stable_Bandwidth_API.md

```diff
@@ -0,0 +1,180 @@
+
+**App-Flow**
+Describes the precise flow the developer wants to prioritize and have stable bandwidth for. This flow is described using source and destination IP addresses, ports and protocols with flow direction.
+
+**Duration**
+Define the number of seconds for which the QoD session should be created. This parameter is optional and if not specified, the session is either deleted on user request or if default expiration limit of 24hrs has been reached.
+
+**Notification URL and token**
+Developers have a chance to specify callback URL on which notifications (eg. session termination) regarding the session can be received from the service provider. This is also an optional parameter.
+
+Sample API invocations are presented in Section 4.5.
+
+## 3\. Authentication and Authorization
+
+The QoD Service API makes use of the client credentials grant which is applicable for server to server use cases involving trusted partners or clients without any protected user data involved.
+In this method the API invoker client is registered as a confidential client with an authorization grant type of client\_credentials [3].
+
+## 4\. API Documentation
+
+### 4.1 Details
+The usage of the QoD throughput API is based on QoS profile classes and parameters which define App-Flows.
+Based on the API, QoS sessions can be created, queried, and deleted. Once a proper QoS profile throughput class is requested, application users get a prioritized service with stable throughput even in the case of congestion.The QoD bandwidth API has the following characteristics:
+* A specified App-Flow is prioritzed to ensure stable throughput for that flow
+* The prioritized App-Flow is described by providing additional information such as protocols, ports, uplink/downlink direction of flow etc. 
+* Stable bandwidth is requested by selecting from the list of QoS profiles made available by the service provider (e.g. Small ,Medium, Large) to map throughput requirements
+* If the developer can optionally specify the duration for which he needs the prioritized App-flow
+* The developer can optionally also specify callback URL on which notifications for the session can be sent
+<br>
+Following diagram shows the interaction between different components 
+
+<br />
+<img src="./resources/QoD_bandwidth_details.PNG" alt="QoD_BM" title="QoD Bandwidth Management"> 
+
+The below table shows sample QoS profiles and are subject to service provider customizations. This sample is taken from the agreed sample (example) set from the Camara-project [2].
+
+| **QoD throughput profile** | **Details** |
+| ---------------------- | ------- |
+| *THROUGHPUT\_S* | Small class of throughput profile - for example DL (Downlink) up-to 10Mbps |
+| *THROUGHPUT\_M* | Medium class of throughput profile - for example DL (Downlink) up-to 30Mbps |
+| *THROUGHPUT\_L* | Large class of throughput profile - for example DL (Downlink) up-to 100Mbps |
```

Comment:
Somewhere in name or details it should mention throughput for DL and UL, as this is more defining for QoD then letters for sizes that can vary (if I correctly understood recent discussions).
When using letters, ideally those letters should have consistent meaning over different networks and future years (might need to keep adding "XL", "2XL"...).

Reply:
The doccumentation is currently in line with API definition, reference implementation code and mapping table. Proposal to discuss changes to this outside of this PR.